addons/mass_mailing/static/src/components/message/message.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="template" xml:space="preserve">
+
+    <t t-inherit="mail.Message" t-inherit-mode="extension">
+        <xpath expr="//t[@t-if][./div/t[@t-if][./span/i[hasclass('o_Message_notificationIcon')][@name='failureIcon']]]" position="attributes">
+            <attribute name="t-if">
+                messageView.messageListViewMessageViewItemOwner and
+                messageView.message.originThread and
+                messageView.message.originThread === messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread
+                and (messageView.message.notifications.length > 0 || messageView.message.traces.length > 0)
```

Comment:
ideally instead of changing template just move condition to a computed field and override the compute method